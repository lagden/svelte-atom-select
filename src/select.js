/* istanbul ignore file */
import Select from './Select.svelte'

export {
	Select
}
