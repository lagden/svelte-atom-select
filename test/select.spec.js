/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Select} from '../src/select.js'

describe('Select', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Select)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Select, {
			props: {
				value: 2,
				class: 'customCss',
				required: true,
				first: 'Selecione',
				options: [
					{value: 1, text: 'Tadashi'},
					{value: 2, text: 'Takamoto'}
				]
			}
		})
		expect(container).toMatchSnapshot()
	})
})
