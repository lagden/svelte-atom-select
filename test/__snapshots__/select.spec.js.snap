// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot default 1`] = `
<body>
  <div>
    <select
      class="_atom_frm__base _atom_frm__select "
    >
      
    </select>
  </div>
</body>
`;

exports[`Select should match snapshot props 1`] = `
<body>
  <div>
    <select
      class="_atom_frm__base _atom_frm__select customCss"
      required=""
    >
      <option
        disabled=""
        value=""
      >
        Selecione
      </option>
      
      <option
        value="1"
      >
        Tadashi
      </option>
      <option
        value="2"
      >
        Takamoto
      </option>
      
    </select>
  </div>
</body>
`;
